package NYT.NClickHouseServer.NProto;

////////////////////////////////////////////////////////////////////////////////

message TReqProcessGossip
{
    required string instance_id = 1;
    required int32 instance_state = 2;
}

message TRspProcessGossip
{
    required string instance_id = 1;
    required int32 instance_state = 2;
}

////////////////////////////////////////////////////////////////////////////////

message TPathWithRevision
{
    required string path = 1;
    required int64 revision = 2;
}

message TReqInvalidateCachedObjectAttributes
{
    repeated TPathWithRevision table_paths = 1;
}

message TRspInvalidateCachedObjectAttributes
{ }

////////////////////////////////////////////////////////////////////////////////

message TSqlObjectInfo
{
    optional uint64 revision = 1;
    optional string create_object_query = 2;
}

////////////////////////////////////////////////////////////////////////////////

message TReqSetSqlObject
{
    optional string object_name = 1;
    optional TSqlObjectInfo object_info = 2;
}

message TRspSetSqlObject
{ }

////////////////////////////////////////////////////////////////////////////////

message TReqRemoveSqlObject
{
    optional string object_name = 1;
    optional uint64 revision = 2;
}

message TRspRemoveSqlObject
{ }

////////////////////////////////////////////////////////////////////////////////
